warp10/src/main/java/io/warp10/script/functions/ECPRIVATE.java

```diff
@@ -66,18 +66,7 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
       throw new WarpScriptException(getName() + " curve name not in " + ECGEN.getCurves() + ".");
     }
 
-    if (!(params.get(Constants.KEY_D) instanceof String)) {
-      throw new WarpScriptException(getName() + " missing or non-String parameter '" + Constants.KEY_D + "'.");
-    }
-
-    String dstr = (String) params.get(Constants.KEY_D);
-    final BigInteger d;
-
-    if (dstr.startsWith("0x") || dstr.startsWith("0X")) {
-      d = new BigInteger(dstr.substring(2), 16);
-    } else {
-      d = new BigInteger(dstr);
-    }
+    final BigInteger d = TOBD.toBigInteger(getName(), params.get(Constants.KEY_D));
```

Comment:
With this function, keys beginning with "OX" are not be processed as before.
Is this intentional?

Reply:
Not sure why this was ever supported. Dropping the support for that awkward syntax is confirmed.